src/utils.js

```diff
@@ -310,22 +299,47 @@ export const setKeychainPin = (pin) => {
 };
 
 /**
- * @params {string} oldPin
- * @params {string} newPin
+ * @param {HathorWallet} wallet
+ * @param {string} oldPin
+ * @param {string} newPin
  *
  * @return {boolean} Wether the change password was successful
  */
-export const changePin = (oldPin, newPin) => {
-  const success = hathorLib.wallet.changePinAndPassword({
-    oldPin,
-    newPin,
-    oldPassword: oldPin,
-    newPassword: newPin,
-  });
-  if (success) {
-    setKeychainPin(newPin);
+export const changePin = async (wallet, oldPin, newPin) => {
+  // We have 2 sets of access data to mutate
+  // The first one is the AsyncStorage access data which is used to start the loaded wallet
+  // The second one is the current loaded wallet so the user can continue to use the app
+  // without needing to restart or close and open.
+
+  const isValidPinWallet = await wallet.checkPin(oldPin);
+  const isValidPinStore = STORE.checkPinAndPasswordOnStore(oldPin);
+  if (!(isValidPinWallet && isValidPinStore)) {
+    return false;
   }
-  return success;
+
+  try {
+    // All of these are checked above so it should not fail
+    await wallet.storage.changePin(oldPin, newPin);
+    let accessData = STORE.getAccessData();
+    accessData = hathorLib.walletUtils.changeEncryptionPin(accessData, oldPin, newPin);
+    accessData = hathorLib.walletUtils.changeEncryptionPassword(accessData, oldPin, newPin);
+    STORE.saveAccessData(accessData);
+  } catch (err) {
+    return false;
+  }
+
+  try {
+    // Will throw if the access data does not have the seed.
```

Comment:
Should we add `@throws` property in the docstring describing the exception and its context?

Reply:
This error is catched